File: Data Structures & Algorithms/three-integer-sum/submission-7.cs
public class Solution {
    public List<List<int>> ThreeSum(int[] nums) {
        List<List<int>> res = new List<List<int>>();
        Array.Sort(nums);
        for(int i = 0; i<nums.Length - 2; i++){
            if(i>0 && nums[i]==nums[i-1]) continue;
            int l = i+1, r = nums.Length-1;
            int num = nums[i];
            while(l<r){
                int sum = num + nums[l] + nums[r];

                if(sum==0){
                    res.Add(new List<int>{num, nums[l], nums[r]});
                    while(l<r && nums[l] == nums[l+1]) l++;
                    while(l<r && nums[r] == nums[r-1]) r--;
                    l++;
                    r--;
                }
                else if(sum>0){
                    r--;
                }
                else{
                    l++;
                }
            }   

        }

        return res;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-7.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = [[] for i in range(len(nums) + 1)]
        count = {}
        result = []
        for num in nums:
            count[num] = count.get(num, 0) + 1
        
        for key, values in count.items():
            freq[values].append(key)
        
        for i in range(len(freq)-1, 0, -1):
            for n in freq[i]:
                result.append(n)
                if(len(result) == k):
                    return result
        
